app/controllers/orderproducts_controller.rb

```diff
@@ -1,6 +1,6 @@
 class OrderproductsController < ApplicationController
   before_action :set_orderproduct, only: [:show, :edit, :update, :destroy]
-
+  before_action :required_user
```

Comment:
have you copied this method from someone in your team?

Reply:
No, it was given in udemy videos to implement url level restrictions.I implemented it using refering that video